src/host/gpunumber.jl

```diff
@@ -0,0 +1,42 @@
+# Custom GPU-compatible `Number` interface.
+struct GPUNumber{T <: AbstractGPUArray} <: AN.AbstractNumber{T}
```

Comment:
I prefer this being written as `AbstractNumbers.`

Then I don't have to remember what `AN` stands for when reading the code.